aws-api/src/main/java/com/amplifyframework/api/aws/AppSyncPageDeserializer.java

```diff
@@ -0,0 +1,103 @@
+package com.amplifyframework.api.aws;
+
+import com.amplifyframework.api.graphql.GraphQLRequest;
+import com.amplifyframework.api.graphql.GraphQLResponse;
+import com.amplifyframework.api.graphql.Page;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.reflect.TypeToken;
+
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+
+/**
+ * Takes an input such as the following and deserializes to an AppSyncPage
+ * {
+ *   "data": {
+ *     "listTodos": {
```

Comment:
Ah, I guess this thing here, the `listTodos` and under. _This_ is the "page" right?

Reply:
This actually was correct based on the design I had implemented yesterday.  I've updated the implementation to the design we discussed last night, which removes the top _two_ levels. (and I have updated this comment).